### PR TITLE
Update API to use the term "dataset"

### PR DESCRIPTION
**Pull request by emmberk**

### CODE REVIEW

**Comment by emmberk** [comment]
❗️Don't merge this. I'm still working out the correct terms with Architecture Team.

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 1]
```diff
@@ -22,12 +22,12 @@ servers:
...
     post:
-      summary: Create a Product
+      summary: Create a dataset
       description: >
-        Clients send a file attachment to create a Product resource
-      operationId: storeProduct
+        Clients send a file attachment to create a dataset.
+      operationId: createDataset
       tags:
         - store
```
Comment (mcalcote) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: should we also update the tag to `create`?

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 2]
```diff
@@ -58,17 +58,17 @@ paths:
         '501':
           $ref: '#/components/responses/NotImplemented'
   # TODO This endpoint will be replaced
-  /product/{productId}/{metadataType}:
+  /dataset/{datasetId}/{metadataType}:
     put:
-      summary: Add metadata to a Product.
+      summary: Add a metadata to a dataset.
```
Comment (mcalcote) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: Add a metadata ***attachment*** to a dataset.

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 3]
```diff
@@ -82,15 +82,15 @@ paths:
       requestBody:
         required: true
         description: >
-          The body of the request that includes one attached file.
+          The body of the request that includes one attached metadata.
```
Comment (mcalcote) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: includes one metadata **_attachment_**

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 4]
```diff
@@ -185,20 +184,20 @@ components:
...
         maxLength: 500
   schemas:
-    File:
+    Attachment:
       type: object
       properties:
         file:
           description: >
-            The file attachment. The current maximum size is the character equivalent of 10GB.
+            The attachment. The current maximum size is the character equivalent of 10GB.
```
Comment (mcalcote) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: dataset attachment? metadata attachment? 
I feel like we need at least one more word to describe what kind of attachment this is...

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml (hunk 1 above)
Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: I'm not sure, but I believe the tag is the API name: https://swagger.io/docs/specification/grouping-operations-with-tags/.

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml (hunk 2 above)
Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: What value does adding the word `attachment` add?

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml (hunk 4 above)
Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: It's both. This schema is used for the both file and metadata attachments. If it's confusing to reuse this schema, I can inline it in the 2 places that it's used.

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml (hunk 2 above)
Comment (mcalcote) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: makes it more readable. only a suggestion

Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: I'm afraid of adding the word attachment here and it becoming a first-level term like `file`, `metadata`, and `dataset`

**Pull request merged**